llm/ggml_test.go

```diff
@@ -0,0 +1,95 @@
+package llm
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+	"text/template"
+)
+
+func TestKVChatTemplate(t *testing.T) {
+	f, err := os.Open(filepath.Join("testdata", "templates.jsonl"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.Close()
+
+	expect := map[string]string{
+		"91aa728ae59c8e30443382f44f48d594b5f7afe91a449c0e5ed0e920a71d60a2": "chatml",
```

Comment:
Instead of a checksum testdata file, could we just put the string in here for less indirection?

Reply:
the jinja2 template doesn't mesh well with go strings. it's easier to store it as a separate file